### PR TITLE
Added a method to create a glob path for leaves files in a directory

### PR DESCRIPTION
**Pull request by etorreborre**
This method is useful to target with a glob all the files which are to be processed under a set of directories, for example `/year=*/month=*/day=*`.

### CODE REVIEW

File: src/main/scala/com/ambiata/poacher/hdfs/Hdfs.scala [hunk 1]
```diff
@@ -182,6 +182,16 @@ object Hdfs {
...
 
+  /**
+   * @return glob string which will glob all the "leaves" files accessible under a given directory
+   */
+  def filesGlob(path: Path): Hdfs[Option[String]] =
+    globFilesRecursively(path).map { files =>
+      files.map(_.depth).maximum.map { max =>
+        Some(List.fill(max - path.depth)("*").mkString("/"))
+      }.getOrElse(None)
+    }
```
Comment (raronson) [inline src/main/scala/com/ambiata/poacher/hdfs/Hdfs.scala]: Does this get all of the leaves, or only the leaves of the largest depth? eg. if you have `/dir/subdir1/file1` and `/dir/file2`, will this return `file1` and `file2` or just `file1`?

Comment (etorreborre) [inline src/main/scala/com/ambiata/poacher/hdfs/Hdfs.scala]: Right, it's not really all the leaves, but "the leaves at the max. depth". I'll change the comment.

Comment (raronson) [inline src/main/scala/com/ambiata/poacher/hdfs/Hdfs.scala]: This seems dangerous though. you might call this on a dir thinking you will get a glob that will catch all of the files, but it actually doesn't, and you wont know what its missed.

**Comment by etorreborre** [comment]
After discussing this PR with @raronson it was decided to drop it: too dangerous and unnecessary.

**Pull request closed without merging**